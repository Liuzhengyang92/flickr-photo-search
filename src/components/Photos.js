import React from 'react';
import Photo from './Photo';

const Photos = ({ photos }) => {
  return (
    <div style={{ display: "flex", flexWrap: "wrap", width: "100%", marginLeft: "8%", marginTop: 100, overflowY: "scroll" }}>
      {!photos.length ? (
        <h1>No Photos to Show...</h1>
      ) : (
        photos.map((photo, index) => {
          return (
              <Photo
                key={index}
                index={index}
                author={photo.author}
                dateTaken={photo.date_taken}
                tags={photo.tags}
                photoLink={photo.link}
                thumbnailPhotoLink={photo.media.m}
              >
              </Photo>
          );
        })
      )}
    </div>
  );
};

export default Photos;