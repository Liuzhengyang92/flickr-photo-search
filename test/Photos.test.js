import React from 'react';
import Photos from '../src/components/Photos';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Photos photos={[]}></Photos>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});