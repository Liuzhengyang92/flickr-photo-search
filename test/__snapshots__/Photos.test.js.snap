// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
      "marginLeft": "8%",
      "marginTop": 100,
      "overflowY": "scroll",
      "width": "100%",
    }
  }
>
  <h1>
    No Photos to Show...
  </h1>
</div>
`;
